accessibility-checker-engine/src/v2/checker/accessibility/util/legacy.ts

```diff
@@ -2653,6 +2653,17 @@ export class RPTUtil {
         return true;
     }
 
+    /**
+     * return true if the node or its ancester is natively hidden or aria-hidden = 'true'
+     * @param node
+     */
+    public static isNodeHidden(node: Element) {
+        if (!RPTUtil.isNodeVisible(node) || node.getAttribute("aria-hidden")) return true;
+        let ancestor = RPTUtil.getAncestor(node, "aria-hidden");
+        if (ancestor && ancestor.getAttribute("aria-hidden")) return true;
+        return false;
+    }
```

Comment:
I believe this function needs more work. `aria-hidden="false"` is valid, but would be treated the same as `aria-hidden="true"` in this function. You have to check `node.getAttribute("aria-hidden") === "true"`. I'm not sure if the field is case sensitive, so you may need to compare against `toLowerCase()`. You can also have a case of:
div[aria-hidden=true]
-- div[aria-hidden=false]
---- div
The second one is ignored, but this code wouldn't get to the one higher in the hierarchy to detect that.